tests/test_smoke.py

```diff
@@ -2947,6 +2947,25 @@ def terminate_replica(replica_id: int) -> str:
     run_one_test(test)
 
 
+@pytest.mark.gcp
+@pytest.mark.sky_serve
+def test_skyserve_base_ondemand_fallback():
```

Comment:
could we add one more tests to simulate preemption? e.g. give a 3 replica deployment with 1 base ondemand replica and we manually terminate one spot. Afterthat we immediately check the service status, and wait until it becomes stable and check status again.

Reply:
added PTAL